Source/engine/dx.cpp

```diff
@@ -236,7 +240,12 @@ void RenderPresent()
 	SDL_Surface *surface = GetOutputSurface();
 
 	if (!gbActive) {
+#ifdef __EMSCRIPTEN__
+		// Just yield to browser when inactive instead of blocking
+		emscripten_sleep(1);

```

Comment:
This is fine for now but long term we'd want to migrate to a single event loop and have the browser drive it (see https://github.com/diasurgical/DevilutionX/issues/3776)

Reply:
<img width="1465" height="1002" alt="image" src="https://github.com/user-attachments/assets/34790a92-0d0e-46e1-84fd-a63fd693cfce" />
This is what happens if you don't have those headers set